### PR TITLE
Adds Orders Summary and Orders Graph

### LINKED ISSUE #410: Orders Report: Summary Numbers
Child task of #168 
[inVision](https://automattic.invisionapp.com/d/main#/projects/prototypes/15060831)

image

API Response [docs](https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_orders_stats):

image

This task involves creation of the Summary Numbers for the Orders Report:

- Number of orders (count)
- Net Revenue
- Avg Order Value
- Refunded

__Questions__
- @LevinMedia it looks like there is some difference between the New Analytics spreadsheet, the API endpoint return, and the design. The design calls for a "Refunded" summary number, but that is not in the API response currently. Average items per order is - should that be shown as a summary number or do we need refunds added to the API response? Answered in comment at p6riRB-3dO-p2 Refunds needed in Summary and Tabular data.

### LINKED ISSUE #417: Order Report: Chart
Child task of #168 
[inVision](https://automattic.invisionapp.com/d/main#/projects/prototypes/15060831)

Add a chart to the Orders Report

image

API Response [docs](https://app.swaggerhub.com/apis/peterfabian/wc-v3-api/1.0.0#/default/get_reports_orders_stats)

### PR DESCRIPTION
**Pull request by belcherj**
Closes #410 and Closes #417 

This adds the Orders Summary #410 and Orders Chart #417 

As stated in #410 the API provides data different that what is in the mock.  This PR adds what is available via the API.

screen shot 2018-10-11 at 3 27 53 pm

### CODE REVIEW

**Comment by belcherj** [comment]
Just realized the Revenue Report has this styled correctly.  Will utilize that summary to git this in tip top.

**Comment by justinshreve** [comment]
@belcherj I'm trying a different approach over here that should possibly work for both of us. https://github.com/woocommerce/wc-admin/pull/510. The current revenue approach feels a bit messy trying to track stuff in state and you can sometimes get caught with the summary numbers infinitely loading.

**Review by timmyc — COMMENTED** [review]
Ah sooo nice to see the chart working on the page! I am wondering though your thoughts on this line comment on the revenue report:

https://github.com/woocommerce/wc-admin/blob/master/client/analytics/report/revenue/index.js#L232

There is quite a bit of code duplication here with the Revenue Report, and while we could continue to copy these patterns across all the reports, it kind of seems like it might be a good time to extract the pattern so we have a single summary number / chart component we can use across a large set of reports.

I know when building out Site Stats in Calypso we ended up duplicating lots of code across many components, just to circle back and DRY it up later - so wondering since we do have some time left in this sprint that we should just take the time and abstract it out now.

Curious to hear others thoughts on this too.

**Comment by psealock** [comment]
>it might be a good time to extract the pattern so we have a single summary number / chart component

I agree. Continuing with this, I like how @Aljullu has pulled out the table in the Orders report to its own file, leaving the report itself free of logic. Its cleaner and makes future DRYing easier.

https://github.com/woocommerce/wc-admin/blob/9a2aa898c8fd7301d13b02155e609181e7060fdd/client/analytics/report/orders/index.js#L32

**Comment by belcherj** [comment]
I agree with this.  Does it make sense to merge then abstract out the code?

**Review by ryelle — APPROVED** [review]
There are a few formatting bugs, but these are things that can be fixed in the next PR.

Average items per order should probably round to an integer?
screen shot 2018-10-12 at 9 59 14 am

The hover card for the average items & order count charts use currency formatting:
screen shot 2018-10-12 at 10 08 14 am

**Comment by ryelle** [comment]
> There is quite a bit of code duplication here with the Revenue Report, and while we could continue to copy these patterns across all the reports, it kind of seems like it might be a good time to extract the pattern so we have a single summary number / chart component we can use across a large set of reports.

Should this be a public component available on `wc.components`, or just something internal we use to keep things DRY? Personally I'm leaning to "internal" (like `analytics/components` for these report-shared components). Maybe we can start with an internal component and move it to wc.components if needed.

**Review by emdashcodes — APPROVED** [review]
Average items per order needs rounded. Approving since that can be fixed in your refactor.

I also noticed that the table does not connect up with the date picker like the revenue report. I know that's separate from this PR but I wanted to note it. cc @Aljullu

**Comment by timmyc** [comment]
> Personally I'm leaning to "internal"

Yep I think that would be a good approach here. We could always export it later if there is a need/desire from the dev community, but for now it feels more like an abstraction for our needs.

File: client/analytics/report/orders/index.js [hunk 1]
```diff
@@ -2,24 +2,239 @@
...
+	getCharts() {
+		return [
+			{
+				key: 'net_revenue',
+				label: __( 'Net Revenue', 'wc-admin' ),
+				type: 'currency',
+			},
+			{
+				key: 'avg_order_value',
+				label: __( 'Avergae Order Value', 'wc-admin' ),
```
Comment (psealock) [inline client/analytics/report/orders/index.js]: 😸 Just noticed this

**Pull request merged**